sound/soc/sof/sof-priv.h

```diff
@@ -401,6 +404,7 @@ struct snd_sof_dev {
 	struct snd_soc_component *component;
 	u32 enabled_cores_mask; /* keep track of enabled cores */
 	struct mutex cores_mask_mutex; /* protects enabled_cores_mask */
+	int core_refs[SND_SOF_CORE_MAX];
```

Comment:
Can we use kref for this?

Reply:
@keyonjie comment not addressed

Reply:
Actually addressed at bottom, let me paste it here:

I previously used that, @plbossart reminded and I agreed that we don't need the heavier kref here as we already have mutex to protect the same resource, the kref itself doesn't help much here for me:

the kref_init() will initialize the refcount to be 1, not 0 that we want.
the kref_get() doesn't support callback, so I still need use kref_read() + check and power up manually.
The kref_put_mutex() can help somewhat here with release callback for power down, but this is the only benefit I can imagine.

So here, I think the light int ref count looks better to me.

@lyakh @plbossart thoughts?